velox/functions/sparksql/Json.cpp

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <functions/prestosql/types/JsonType.h>
+#include "velox/expression/VectorWriters.h"
+
+namespace facebook::velox::functions::sparksql {
+namespace {
+
+class ToJsonFunction : public exec::VectorFunction {
+ public:
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& outputType,
+      exec::EvalCtx& context,
+      VectorPtr& result) const override {
+    JsonCastOperator::get()->castTo(
+        *args.at(0), context, rows, outputType, result);
+  }
+
+  static std::vector<std::shared_ptr<exec::FunctionSignature>> signatures() {
+    return {
+        exec::FunctionSignatureBuilder()
+            .typeVariable("T")
+            .returnType("varchar")
+            .argumentType("T")
```

Comment:
To align with spark, it looks only array type & struct (map) type are acceptable. If some inner types like Date type, Timestamp type, etc. are not actually supported, we should not cover them in function signature.

Reply:
Yes, you're right, only array, struct(row), and map type are acceptable in Spark. I used to try to add this constriction in the function signature. But I didn't find a way to add a signature for row type. It seemed that we must specify each column's type for row type. I tried `.argumentType("row")`, `argumentType("row(T)")` and `argumentType("row(T...)"` but  none work.

Reply:
@PHILO-HE PH FYI, https://github.com/facebookincubator/velox/discussions/6403

Reply:
OK, let's keep your current impl. Thanks for clarification!